### PR TITLE
Add a shim lib for external modules

### PR DESCRIPTION
**Pull request by busterb**
This adds a small shim library to make adding external modules simpler. This removes some implementation details from how an external module interface from the module itself.

## Verification Steps

 - [x] Ensure that 'msfconsole' starts as expected without any module exceptions when running on a platform where python is installed
 - [x] Verify that 'exploits/linux/smtp/haraka' continues working as expected.

### CODE REVIEW

**Comment by busterb** [comment]
It would be pretty easy to add a ruby equivalent too for testing external ruby modules. Note, this is just a stepping stone for how we eventually want this to look, which will continue as we learn more.

File: lib/msf/core/modules/external/bridge.rb [hunk 1]
```diff
@@ -36,12 +36,16 @@ def get_status
   def initialize(module_path)
     self.running = false
     self.path = module_path
+    self.env = {
+      'PYTHONPATH' => ENV['PYTHONPATH'] + File::PATH_SEPARATOR + File.expand_path('../python', __FILE__)
```
Comment (acammack-r7) [inline lib/msf/core/modules/external/bridge.rb]: Adding `env` to the class is a good idea, but I think adding this line would be a better fit in a class that looked like:
```ruby
class Msf::Core::Modules::External::PyBridge < Msf::Core::Modules::External::Bridge
  def self.applies?(module_name)
    module_name.match? /\.py$/
  end

  def initialize(module_path)
    super
    self.env = self.env.merge {'PYTHONPATH' => ENV['PYTHONPATH'] + File::PATH_SEPARATOR + File.expand_path('../python', __FILE__)}
  end
end
```

And adding `Bridge.open` that auto-chooses the most applicable class. This approach will later enable us to add bits that will construct the correct invocation on Windows, or even use compiled languages (would require a compiler, though).

Comment (busterb) [inline lib/msf/core/modules/external/bridge.rb]: ok, I'll work on splitting this out.

**Comment by wvu-r7** [comment]
Thanks for grabbing this, @acammack-r7.

**Comment by acammack-r7** [comment]
ba9010730ae5a638d2a2dedb472b0448ef381e41

**Comment by acammack-r7** [comment]
Release Notes
===========

Code common to writing Python modules is now abstracted out and available with `from metasploit import module`.

**Comment by busterb** [comment]
Thanks @acammack-r7 !

**Pull request merged**